syntax = "proto3";

package challenge;

service Handler {
  rpc SayString(StringRequest) returns (StringReply) {}
  rpc SayNumber(NumberRequest) returns (NumberReply) {}
  rpc ReceiveFile(stream FileChunk) returns(FileAck) {}
}

message StringRequest {
  string str = 1;
}

message StringReply {
  string message = 1;
}

message NumberRequest {
  int32 num = 1;
}

message NumberReply {
  int32 message = 1;
}

message FileChunk {
  string    FileName = 1;
  int64     SizeInBytes = 2;
  bytes     ControlSum = 3;
  bytes     Data = 4;
  int64     ChunkNumber = 5;
  bool      IsLastChunk = 6;
}

message FileAck {
  string    FileName = 1;
  int64     SizeInBytes = 2;
  bytes     ControlSum = 3;
  string    Error = 4;
}
